web/components/SimpleCallToAction/SimpleCallToAction.tsx

```diff
@@ -2,13 +2,15 @@ import { SimpleCallToAction as SimpleCallToActionProps } from '../../types/Simpl
 import ButtonLink from '../ButtonLink';
 import urlJoin from 'proper-url-join';
 
-export const SimpleCallToAction = ({
-  text,
-  url,
-  reference,
-}: SimpleCallToActionProps) =>
-  text && reference?.slug?.current ? (
-    <ButtonLink text={text} url={urlJoin(reference?.slug?.current)} />
-  ) : text && url ? (
-    <ButtonLink text={text} url={url} />
+export const SimpleCallToAction = ({ text, link }: SimpleCallToActionProps) => {
+  const openInNewTab = Boolean(link?.blank);
+  return text && link?.internal?.slug?.current ? (
+    <ButtonLink
+      text={text}
+      url={urlJoin(link.internal.slug.current)}
+      openInNewTab={openInNewTab}
+    />
+  ) : text && link?.external ? (
+    <ButtonLink text={text} url={link.external} openInNewTab={openInNewTab} />
   ) : null;
+};
```

Comment:
Nit: is this any easier to read? Or some other variant eliminating the nested ternary.

```suggestion
export const SimpleCallToAction = ({ text, link }: SimpleCallToActionProps) =>
  !text || !link || !(link.internal?.slug?.current || link.external) ? null : (
    <ButtonLink
      text={text}
      openInNewTab={Boolean(link?.blank)}
      url={link.external || urlJoin(link.internal?.slug?.current)}
    />
  );
```

Reply:
Personally I find that alternative harder to read.

Would you prefer a sequence of returns, e.g. like this?

```suggestion
export const SimpleCallToAction = ({ text, link }: SimpleCallToActionProps) => {
  const openInNewTab = Boolean(link?.blank);
  if (text && link?.internal?.slug?.current) {
    return (
      <ButtonLink
        text={text}
        url={urlJoin(link.internal.slug.current)}
        openInNewTab={openInNewTab}
      />
    );
  }
  if (text && link?.external) {
    return (
      <ButtonLink text={text} url={link.external} openInNewTab={openInNewTab} />
    );
  }
  return null;
};
```

Another alternative could be to `let url = null;` along with an if/elseif assignment instead of doing a single assignment to a nested/chained ternary expression. I personally don't mind `let`, I just try to avoid them when possible because I got the impression that some other people prefer const.

I anticipate another upcoming variation between internal and external here, by the way, related to the arrow styling. Might affect the readability of the various alternatives one way or the other.

Reply:
I like your "suggested" suggestion, which also seems more malleable wrt. future iterations. :slightly_smiling_face: (But again, this is a nit, feel absolutely free to go for whichever version you prefer.)